### PR TITLE
refactor: modify credentials from map to object

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
Use Credential and MqttCredential object instead of Map. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/device/session/MqttSessionFactory.java [hunk 1]
```diff
@@ -22,16 +23,15 @@ public MqttSessionFactory(IotAuthClient iotAuthClient) {
     }
 
     @Override
-    public Session createSession(Map<String, String> credentialMap) throws AuthenticationException {
-        // TODO: replace with jackson object mapper
-        MqttCredential mqttCredential = new MqttCredential(credentialMap);
+    public Session createSession(Credential credential) throws AuthenticationException {
+        MqttCredential mqttCredential = (MqttCredential) credential;
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/device/session/MqttSessionFactory.java:27]: do not blindly cast, make sure it is the right type first

**Review by MikeDombo — COMMENTED** [review]
What is the motivation for this change?

**Comment by MikeDombo** [comment]
I do not think that we want this because it seems to force the credential object creation into IPC, rather than abstracting it and letting the session implementation handle it on its own. For example, if we wanted to support some new auth scheme then the current interface allows for scheme (ex. mqtt) registration and then the new authenticator can parse the input in a way that it understands.

File: src/main/java/com/aws/greengrass/device/session/credentials/Credential.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.aws.greengrass.device.session.credentials;
```
Comment (jbutler) [inline src/main/java/com/aws/greengrass/device/session/credentials/Credential.java:1]: copyright

File: src/main/java/com/aws/greengrass/device/session/credentials/MqttCredential.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.aws.greengrass.device.session.credentials;
```
Comment (jbutler) [inline src/main/java/com/aws/greengrass/device/session/credentials/MqttCredential.java:1]: copyright

File: src/main/java/com/aws/greengrass/device/session/SessionCreator.java [hunk 4]
```diff
@@ -22,28 +23,23 @@ private SessionCreator() {
...
             logger.atWarn().kv("credentialType", credentialType)
-                .log("no registered handler to process device credentials");
+                    .log("no registered handler to process device credentials");
             throw new IllegalArgumentException("unknown credential type");
         }
-        return sessionFactory.createSession(credentialMap);
+        return sessionFactory.createSession(credential);
     }
 
     public static void registerSessionFactory(String credentialType, SessionFactory sessionFactory) {
```
Comment (jbutler) [inline src/main/java/com/aws/greengrass/device/session/SessionCreator.java:45]: Credential registration will break with this change. Please update PR description once tested

**Pull request closed without merging**